### ISSUE TITLE
Why is CapOptions.UseDispatchingPerGroup obsolete?

### ISSUE THREAD

**Issue by TipluSebastian** [dotnetcore/CAP#1385]
Why is `CapOptions.UseDispatchingPerGroup` obsolete? I still need to have an independent dispatching pipeline for each message group. Using `CapOptions.EnableConsumerPrefetch` as stated in the obsolete message does not have the desired effect.

**Comment by yang-xiaodong** [comment]
In the past, our consumers were executed linearly (OneByOne). At times, when consumers were too slow, it would impact subsequent executions. Therefore, by introducing UseDispatchingPerGroup, we created a queue for each group and dispatched consumers to each queue by group, controlling the process through parallel execution between groups. Now, when EnableConsumerPrefetch is enabled, consumers are scheduled by the .NET thread pool and each consumer is executed in parallel. As such, UseDispatchingPerGroup is no longer needed.

**Comment by TipluSebastian** [comment]
> In the past, our consumers were executed linearly (OneByOne). [...] by introducing UseDispatchingPerGroup, we created a queue for each group and dispatched consumers to each queue by group, controlling the process through parallel execution between groups.

I need exactly this behavior. Could you please add an additonal option to opt-out of the new behavior?

**Comment by yang-xiaodong** [comment]
@TipluSebastian Can you describe why you need this behavior, what's the scenario?

**Comment by yang-xiaodong** [comment]
Closed by https://github.com/dotnetcore/CAP/issues/1399

**Comment by jerviscui** [comment]
When one queue generates a large number of messages and fills the Channel, other queues cannot be consumed in parallel.

**Comment by jerviscui** [comment]
> [@jerviscui](https://github.com/jerviscui) See [#1537](https://github.com/dotnetcore/CAP/pull/1537)  查看 [#1537](https://github.com/dotnetcore/CAP/pull/1537)

I think we are talking about two questions.
1. What https://github.com/dotnetcore/CAP/pull/1537 does is limit the consumption rate of a particular queue.
2. And what I'm trying to say is that a certain queue generating a lot of messages will fill up the Channel, causing other queues to be unable to write the Channel to be consumed.

**Comment by yang-xiaodong** [comment]
@jerviscui In #1537, we introduced **GroupConcurrent**, which means that only messages with configured values will be pulled into memory for parallel consumption — eliminating the possibility of channels being full.

For example, if you want to consume messages from two topics in parallel, simply assign these two topics to different groups. This is different from the past, and now messages will be backlogged to Broker.

see https://www.cnblogs.com/savorboard/p/18268210/cap-8-2 for more info.

**Comment by jerviscui** [comment]
Hi, @yang-xiaodong 
I understand that GroupConcurrent limits the rate of parallel consumption per queue. My problem is that all queues are sharing a Channel:

https://github.com/dotnetcore/CAP/blob/b4095625dd9aa36bbc2e8c3e62026b0461e72cc0/src/DotNetCore.CAP/Processor/IDispatcher.Default.cs#L186-L198

For example, 

```csharp
[CapSubscribe("hello")]
public void Hello1(){
  Console.WriteLine("hello 1");
}

[CapSubscribe("hello2")]
public void Hello2(){
  Console.WriteLine("hello 2");
}
```

Assuming that 10,000 hello events are generated now, the Channel will be filled, and hello2 will not be consumed until the Channel has space.

**Comment by yang-xiaodong** [comment]
UseDispatchingPerGroup is for the group, you're not  specify group for consumers `Hello1` and `Hello2`, they are same group by default. @jerviscui
